source/server/listener_manager_impl.cc

```diff
@@ -75,19 +75,23 @@ ProdListenerComponentFactory::createListenerFilterFactoryList_(
 Network::SocketSharedPtr
 ProdListenerComponentFactory::createListenSocket(Network::Address::InstanceConstSharedPtr address,
                                                  bool bind_to_port) {
-  // For each listener config we share a single TcpListenSocket among all threaded listeners.
   // UdsListenerSockets are not managed and do not participate in hot restart as they are only
-  // used for testing. First we try to get the socket from our parent if applicable.
-  // TODO(mattklein123): UDS support.
-  ASSERT(address->type() == Network::Address::Type::Ip);
+  // used for testing.
+  ASSERT(address->type() == Network::Address::Type::Ip ||
+         address->type() == Network::Address::Type::Pipe);
+  if (address->type() == Network::Address::Type::Pipe) {
+    return std::make_shared<Network::UdsListenSocket>(address->asString());
```

Comment:
Quick non-full review comment: This is not going to support hot restart. I think it should be really easy to support hot restart by folding this logic into the hot restart duplication logic below. We already pass a string for the address name to the parent process, where we can then look up the socket by name. I'm pretty sure that it will mostly just work if you add code below to handle creating a "unix://" socket name and passing it. 

Also, we should definitely have an integration test for this, and optimally add a unix socket to the hot restart test also. @ggreenway or @alyssawilk can likely guide you.

Reply:
OK, updated this code to use the same hot-restart recovery logic for UDS as is used for TCP.